src/pages/steps/AdditionalInformationStep/PriceInformation.tsx

```diff
@@ -153,95 +142,75 @@ const PriceInformation = ({
   );
 
   // @ts-expect-error
-  const offer: Event | Place | undefined = getOfferByIdQuery.data;
+  const offer: Offer | undefined = getOfferByIdQuery.data;
 
   const {
     register,
     trigger,
     control,
     setValue,
-    getValues,
     handleSubmit,
     formState: { errors },
-    watch,
   } = useForm<FormData>({
-    mode: 'onBlur',
-    reValidateMode: 'onBlur',
     resolver: yupResolver(schema),
+    defaultValues: defaultPriceInfoValues,
     shouldFocusError: false,
-    defaultValues: { rates: [] },
   });
 
-  const errorRates = useMemo(
-    () => (errors?.rates ?? []).filter((error: any) => error !== undefined),
-    [errors.rates],
-  );
-
-  const hasGlobalError = useMemo(() => errorRates.length > 0, [errorRates]);
-  const duplicateNameError = useMemo(
-    () => errorRates.find((error) => error.type === 'uniqueName')?.message,
-    [errorRates],
-  );
-  const hasUitpasError = useMemo(
-    () => errorRates.some((error) => error.name?.type === 'name-is-not-uitpas'),
-    [errorRates],
-  );
+  const { fields, replace, append, remove } = useFieldArray({
+    name: 'rates',
+    control,
+  });
 
-  const ratesField = useFieldArray({ name: 'rates', control });
-  const rates = watch('rates');
-  const controlledRates = ratesField.fields.map((field, index) => ({
-    ...field,
-    ...rates[index],
-  }));
+  const rates = fields;
```

Comment:
I think `fields` is only the default value not the current value so I don't think you can infer `hasUitpasPrices` from it no?

![Screenshot 2023-02-10 at 12 02 24](https://user-images.githubusercontent.com/1321596/218076350-f6150bd9-3295-4825-b6cd-b792456a2c72.png)